velox/functions/prestosql/Bitwise.h

```diff
@@ -95,67 +95,79 @@ struct BitwiseArithmeticShiftRightFunction {
   }
 };
 
-namespace {
-template <typename T, int MAX_SHIFT>
+namespace detail {
+template <typename T>
 #if defined(__clang__)
 __attribute__((no_sanitize("integer")))
 #endif
-FOLLY_ALWAYS_INLINE bool
-bitwiseLeftShift(int64_t& result, T number, T shift) {
-  if ((uint64_t)shift >= MAX_SHIFT) {
-    result = 0;
+FOLLY_ALWAYS_INLINE T
+bitwiseLeftShift(T number, int32_t shift) {
+  static constexpr uint32_t kMaxShift = sizeof(T) * 8;
+  if ((uint32_t)shift >= kMaxShift) {
+    return 0;
   } else {
-    result = (number << shift);
+    return (number << shift);
   }
-  return true;
 }
-} // namespace
+} // namespace detail
 
-template <typename T>
+template <typename TExec>
 struct BitwiseLeftShiftFunction {
-  template <typename TInput>
-  FOLLY_ALWAYS_INLINE bool call(int64_t& result, TInput number, TInput shift) {
-    if constexpr (std::is_same_v<TInput, int8_t>) {
-      return bitwiseLeftShift<TInput, 8>(result, number, shift);
-    } else if constexpr (std::is_same_v<TInput, int16_t>) {
-      return bitwiseLeftShift<TInput, 16>(result, number, shift);
-    } else if constexpr (std::is_same_v<TInput, int32_t>) {
-      return bitwiseLeftShift<TInput, 32>(result, number, shift);
-    } else {
-      return bitwiseLeftShift<TInput, 64>(result, number, shift);
-    }
+  FOLLY_ALWAYS_INLINE void
+  call(int64_t& result, int64_t number, int32_t shift) {
+    result = detail::bitwiseLeftShift<int64_t>(number, shift);
+  }
+
+  FOLLY_ALWAYS_INLINE void
+  call(int32_t& result, int32_t number, int32_t shift) {
+    result = detail::bitwiseLeftShift<int32_t>(number, shift);
+  }
+
+  FOLLY_ALWAYS_INLINE void
+  call(int16_t& result, int16_t number, int32_t shift) {
+    result = detail::bitwiseLeftShift<int16_t>(number, shift);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(int8_t& result, int8_t number, int32_t shift) {
+    result = detail::bitwiseLeftShift<int8_t>(number, shift);
   }
 };
 
-namespace {
-template <typename T, typename UT, int MAX_SHIFT>
-FOLLY_ALWAYS_INLINE bool bitwiseRightShift(int64_t& result, T number, T shift) {
-  if ((uint64_t)shift >= MAX_SHIFT) {
-    result = 0;
+namespace detail {
+template <typename T, typename UT>
```

Comment:
seems like these two types are always the same?

Reply:
It might be a bit hard to see, but there are signed and unsigned types: int64_t, uint64_t.